### ISSUE TITLE
[print] improve ModuleList print format

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#8900]
currently:
```py
>>> nn.ParameterList([nn.Parameter(torch.randn(3))])
ParameterList(
    (0): Parameter containing: [torch.FloatTensor of size 3]

)
```

**Comment by karandwivedi42** [comment]
@SsnL can you post an example of the new format? I will PR this.

**Comment by SsnL** [comment]
@karandwivedi42 Sorry I don't know how it should be properly done too. The current expr looks more like the old tensor print fmt. @li-roy might have opinion on how these things should look.

**Comment by apaszke** [comment]
I think what we have now is reasonable. We don't want the full repr there, because it's more important to print the structure of the module, and not the exact weight values (which are rarely interpretable)

**Comment by SsnL** [comment]
@apaszke Yeah, we shouldn't print the values. But we don't show classes like `torch.FloatTensor` anymore, and explicit using these classes are discouraged. We should show `dtype` and `device` instead.

We should remove the extra line either way.
